app/server.py

```diff
@@ -1,12 +1,88 @@
-from sanic import Sanic, text, request
+from configparser import ConfigParser
+from pathlib import Path
+import random
+from typing import Union
 
-app = Sanic(name="api")
+from kink import inject
+from sanic import Sanic, json
 
+from app.services import interfaces, configure_services
+from app.utils import get_app_config
 
-@app.route("/")
-async def home(request):
-    return text("Hello World")
 
+class ProtocolError(Exception):
+    """
+    Raised where an implementation does not meet the required protocols.
+    """
 
-if __name__ == "__main__":
-    app.run(host="0.0.0.0", port=3000, debug=True)
+    def __init__(self, msg: str):
+        self.msg = msg
+
+
+@inject
```

Comment:
I'd be tempted to inject the store/messenger implementations further down in code. For instance we could inject the messenger service only to API endpoints which require it. 

That way, any setup that needs to happen can occur in the `__init__` function and we know that wherever we consume it, we don't have to do any setup, but we also know that the setup is never done until we're sure we need to use it (lazy loading).

Reply:
good shout, done in latest changes 👍